bed/bedPe/bedPe.go

```diff
@@ -0,0 +1,169 @@
+		for i := range bunk.A.Annotation {
+			out = fmt.Sprintf("%s\t%s", out, bunk.A.Annotation[i])
+		}
+		return out
+	default:
+		log.Fatalf("Error: expecting a request to print at least 6 bedPe fields, but got: %d\n", fields)
+	}
+	return ""
+}
+
+// WriteToFileHandle writes an input BedPe struct to an io.Writer
+func WriteToFileHandle(file io.Writer, rec BedPe) {
+	var err error
+	_, err = fmt.Fprintf(file, "%s\n", rec)
+	exception.PanicOnErr(err)
+}
+
+//Write writes a slice of BedPe structs to a specified filename.
+func Write(filename string, records []BedPe) {
+	var err error
+	file := fileio.EasyCreate(filename)
+
+	for i := range records {
+		WriteToFileHandle(file, records[i])
+	}
+	err = file.Close()
+	exception.PanicOnErr(err)
+}
+
+// Read returns a slice of BedPe structs from an input filename.
+func Read(filename string) []BedPe {
+	var line string
+	var answer []BedPe
+	var err error
+	var doneReading bool
+
+	file := fileio.EasyOpen(filename)
+
+	for line, doneReading = fileio.EasyNextRealLine(file); !doneReading; line, doneReading = fileio.EasyNextRealLine(file) {
+		current := processBedPeLine(line)
+		answer = append(answer, current)
+	}
+	err = file.Close()
+	exception.PanicOnErr(err)
+	return answer
+}
+
+//processBedPeLine is a helper function of Read that returns a BedPe struct from an input line of a file.
+func processBedPeLine(line string) BedPe {
+	var startANum, endANum, startBNum, endBNum int
+	words := strings.Split(line, "\t")
+	startANum = common.StringToInt(words[1])
+	endANum = common.StringToInt(words[2])
+	startBNum = common.StringToInt(words[4])
+	endBNum = common.StringToInt(words[5])
+
+	current := BedPe{A: bed.Bed{
+		Chrom: words[0],
+		ChromStart: startANum,
+		ChromEnd: endANum,
+		Strand: bed.None,
+		FieldsInitialized: len(words),
+		},
+		B: bed.Bed{
+		Chrom: words[3],
+		ChromStart: startBNum,
+		ChromEnd: endBNum,
+		Strand: bed.None,
+		FieldsInitialized: len(words),
+		},
+	}
+	if len(words) >= 7 {
+		current.A.Name, current.B.Name = words[6], words[6]
+	}
+	if len(words) >= 8 {
+		current.A.Score, current.B.Score = common.StringToInt(words[7]), common.StringToInt(words[7])
+	}
+	if len(words) >= 9 {
+		current.A.Strand = bed.StringToStrand(words[8])
+	}
+	if len(words) >= 10 {
+		current.B.Strand = bed.StringToStrand(words[9])
+	}
+	if len(words) >= 11 {
+		for i := 10; i < len(words); i++ {
+			current.A.Annotation = append(current.A.Annotation, words[i])
+			current.B.Annotation = append(current.B.Annotation, words[i])
+		}
+	}
+
+	return current
+}
+
+// NextBedPe returns a BedPe struct from an input fileio.EasyReader. Returns a bool that is true when the reader is done.
+func NextBedPe(reader *fileio.EasyReader) (BedPe, bool) {
+	line, done := fileio.EasyNextLine(reader)
```

Comment:
This will not allow comment lines in the file, but changing to "EasyNextRealLine" will allow comment lines